javaparser-core/src/main/java/com/github/javaparser/resolution/types/ResolvedReferenceType.java

```diff
@@ -406,18 +404,9 @@ public boolean isRawType() {
     }
 
     public List<ResolvedMethodDeclaration> getAllMethodsVisibleToInheritors() {
-        List<ResolvedMethodDeclaration> res = new LinkedList<>(this.getDeclaredMethods().stream()
-                .map(m -> m.getDeclaration())
+        return this.getAllMethods().stream()
                 .filter(m -> m.accessSpecifier() != PRIVATE)
-                .collect(Collectors.toList()));
-
-        // We want to avoid infinite recursion in case of Object having Object as ancestor
-        if (!(Object.class.getCanonicalName().equals(getQualifiedName()))) {
```

Comment:
Are we sure this is not necessarily anymore?

Reply:
I think so because:
- the getAllMethods() didn't had that safety-check
- the implementation of getDirectAncestors() seems to handle the Object.class ancestor already

Reply:
Then it looks good to me